### DISCUSSION TITLE
Passing environment variables to the Docker Image.

### DISCUSSION THREAD

**Discussion by mbarinov** [vercel/next.js#25236] [Help]
What are the best practices for passing env variables to Docker Image? 

I've tried to set up them in Dockerfile. As a result, I couldn't get them in next.config.js all items are undefined. How do you pass envs to the Next.js app?

```module.exports = {
  future: {
    webpack5: true,
  },
  publicRuntimeConfig: {
    apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
    authDomain: process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN,
    projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID,
    appID: process.env.NEXT_PUBLIC_FIREBASE_APP_ID,
  },
};```

**Comment by bobarros** [comment]
With Dockerfile, you can use ENV PORT 3000 to set a variable PORT with the value 3000, for example.

After that, a local .env file at your next project directory will do the job. 

If it is available to you, might be easy to use Compose and load the .env file with all your variables.

**Reply by mbarinov** [reply]
Yeah, of course. The problem is how to configure an app in the cloud. For example, I need to pass ENV variables from cloud settings to the app in a Docker image. Maybe you know how to do this?

**Comment by bobarros** [comment]
Either I have no idea how to help with this matter or I do not understand what you need when you say in the title "Passing environment variables to the Docker Image"

In both situations, I can't help. Sorry about that.
I hope that some else might do it.

**Comment by wwwdepot** [comment]
Lookup your cloud provider docs on how to set environment variables. Google search: **_CLOUD_PROVIDER_ docker environment variables** should give you plenty of results

- https://docs.microsoft.com/en-us/azure/container-instances/container-instances-environment-variables
- https://cloud.google.com/compute/docs/containers/configuring-options-to-run-containers#setting_environment_variables
